packages/make-styles/src/runtime/resolveStyleRules.ts

```diff
@@ -28,9 +51,10 @@ export function resolveStyleRules(
   pseudo = '',
   media = '',
   support = '',
-  result: Record<string, MakeStylesResolvedRule> = {},
+  cssClassesMap: CSSClassesMap = {},
+  cssRulesByBucket: CSSRulesByBucket = {},
   rtlValue?: string,
-): Record<string, MakeStylesResolvedRule> {
+): [CSSClassesMap, CSSRulesByBucket] {
   const expandedStyles: MakeStyles = expandShorthand(resolveProxyValues(styles));
```

Comment:
`resolveStyleRules` is called recursively on nested objects, `expandShorthand` goes deep into styles = we run expand multiple times on nested objects 🐛

Reply:
The same for `resolveProxyValues`

Reply:
Good catch 😎 I will implement it with your suggestion (https://github.com/microsoft/fluentui/pull/18169#discussion_r659581002).

Reply:
Done in 3c6c50a.